plugin/src/plugin.rs

```diff
@@ -57,11 +57,17 @@ impl<'a, T: 'a + Clone> Plugin<T> {
 
     pub fn log(&self, level: LogLevel, msg: &str) -> &Self {
         let mut writer = io::stdout();
-        let mut log_req = init_success_response(40);
-        add_str(&mut log_req, "level", &level.to_string()[0..]);
-        add_str(&mut log_req, "message", msg);
+        let mut payload = init_payload();
+        add_str(&mut payload, "level", &level.to_string()[0..]);
+        add_str(&mut payload, "message", msg);
+        let request = Request {
+            id: None,
```

Comment:
To start with, it does not log on my system. The plugin dies on startup, I can confirm as when I try to call the RPC method, it returns with this error message:
```
{
   "code": -4,
   "message": "Plugin terminated before replying to RPC call."
}
```
And on subsequent RPC method calls, it returns:
```
{
   "code": -32601,
   "message": "Unknown command 'hello'"
}
```
So naturally I check the log, and this is the message with which it terminates:
`2022-06-25T07:13:35.537Z UNUSUAL plugin-foo_plugin: Killing plugin: JSON-RPC response "id"-field is not a u64`
 I start debugging the issue, and realize that we have the `id` field of `Request` inside the `common/src/types.rs` as `Option<u64>` and we use a `None` when creating the payload inside the `log()`. Not sure if `None` is a valid `id`. A simple fix was to use `Some(0)` as the `id` field (might be wrong but i had to try something). Which tho successfully compiles but terminates on startup like earlier, but with a different message this time (practically the same issue as yesterday with the Dart plugin) 
`2022-06-25T07:28:52.429Z UNUSUAL plugin-foo_plugin: Killing plugin: Received a JSON-RPC response for non-existent request`

Reply:
Good catch, thanks you find a bug in my implementation, because the log is a notification, and the notification in JSON RPC 2.0 is without an id, so I missed in the common package the serde json skip if is none!

Pushed the fix, and lets see if the CI is happy now :)